### ISSUE TITLE
Apple Silicon (darwin/arm64) support

### ISSUE THREAD

**Issue by mzanini** [heroku/terraform-provider-heroku#301]
### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

### Heroku Provider Version
```
❯ terraform -v
Terraform v0.14.9
+ provider registry.terraform.io/heroku/heroku v4.1.0
```

### Debug Output
I figured the output was small enough to be pasted here:
```
❯ terraform init
.......
Error: Incompatible provider version

Provider registry.terraform.io/heroku/heroku v4.1.0 does not have a package
available for your current platform, darwin_arm64.
```

### Expected Behavior
`terraform init` initializes the provider without errors

### Actual Behavior
`terraform init` returned an error "Incompatible provider version"

### Steps to Reproduce
1. on a new Mac with Apple M1 processor, try to initialize terraform configuration that uses a Heroku provider
2. See that the initialization fails with the message posted above

### Important Factoids
Running on a Mac Mini with Apple Silicon

### References
- [darwin/arm64 support issue in the terraform repo](https://github.com/hashicorp/terraform/issues/27257)

**Comment by davidji99** [comment]
Hi @mzanini,

Could you try out `v4.1.1-beta` on your M1 apple machine?

**Comment by davidji99** [comment]
This is fixed from `v4.1.1` going forward.

**Comment by mzanini** [comment]
Sorry I didn't see the notification. Confirmed that it works with version `4.1.1`. Thanks so much!!
